src/WindowsUtils/include/WindowsUtils/Debugger.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef WINDOWS_UTILS_DEBUGGER_H_
+#define WINDOWS_UTILS_DEBUGGER_H_
+
+// clang-format off
+#include <Windows.h>
+#include <psapi.h>
+// clang-format on
+
+#include "OrbitBase/Promise.h"
+#include "OrbitBase/Result.h"
+#include "WindowsUtils/CreateProcess.h"
+
+#include <filesystem>
+#include <string>
+#include <thread>
+#include <vector>
+
+namespace orbit_windows_utils {
+
+struct DebugEventListener {
+  virtual void OnCreateProcessDebugEvent(const DEBUG_EVENT& event) = 0;
+  virtual void OnExitProcessDebugEvent(const DEBUG_EVENT& event) = 0;
+  virtual void OnCreateThreadDebugEvent(const DEBUG_EVENT& event) = 0;
+  virtual void OnExitThreadDebugEvent(const DEBUG_EVENT& event) = 0;
+  virtual void OnLoadDllDebugEvent(const DEBUG_EVENT& event) = 0;
+  virtual void OnUnLoadDllDebugEvent(const DEBUG_EVENT& event) = 0;
+  virtual void OnBreakpointDebugEvent(const DEBUG_EVENT& event) = 0;
+  virtual void OnOutputStringDebugEvent(const DEBUG_EVENT& event) = 0;
+  virtual void OnExceptionDebugEvent(const DEBUG_EVENT& event) = 0;
+  virtual void OnRipEvent(const DEBUG_EVENT& event) = 0;
+};
+
+class Debugger {
```

Comment:
Nit: Could you add a class level comment? Especially the intended usage should be documented.